test/unit/specs/field-types/TypedFieldComponent.spec.js

```diff
@@ -0,0 +1,303 @@
+    expect(TypedFieldComponent.name).to.equal('TypedFieldComponent')
+  })
+
+  it('renders correctly with minimal props', () => {
+    expect(wrapper.contains('input')).to.equal(true)
+    expect(wrapper.contains('label')).to.equal(true)
+    expect(wrapper.contains('small')).to.equal(true)
+  })
+
+  it('should load default data with the help of props', () => {
+    expect(wrapper.vm.localValue).to.equal('hallo')
+  })
+
+  it('should render the label correctly', () => {
+    expect(wrapper.contains('label')).to.equal(true)
+    const label = wrapper.find('label')
+    expect(label.text()).to.equal('Test Field')
+    expect(label.element.htmlFor).to.equal('test-field')
+  })
+
+  it('should render the description correctly', () => {
+    expect(wrapper.contains('small')).to.equal(true)
+    const description = wrapper.find('small')
+    expect(description.text()).to.equal('This is a test field')
+    expect(description.element.id).to.equal('test-field-description')
+    expect(description.element.className).to.equal('form-text text-muted')
+  })
+
+  it('should render the input correctly', () => {
+    expect(wrapper.contains('input')).to.equal(true)
+    const input = wrapper.find('input').element
+    expect(input.id).to.equal('test-field')
+    expect(input.name).to.equal('test-field')
+    expect(input.type).to.equal('text')
+    expect(input.required).to.equal(true)
+    expect(input.className).to.equal(
+      'form-control form-control-lg vf-pristine vf-invalid vf-untouched vf-invalid-validate')
+  })
+
+  it('should emit an updated value on change', () => {
+    wrapper.setData({localValue: 'test'})
+    expect(wrapper.emitted().input[0]).to.deep.equal(['test'])
+
+    wrapper.setData({localValue: 'test another'})
+    expect(wrapper.emitted().input[1]).to.deep.equal(['test another'])
+  })
+
+  it('should receive the "is-invalid" class if not valid', () => {
+    wrapper.setData({
+      state: {
+        $touched: true,
+        $invalid: true
+      }
+    })
+
+    expect(wrapper.find('input').classes()).to.deep.equal(['form-control', 'form-control-lg', 'is-invalid', 'vf-pristine', 'vf-invalid', 'vf-untouched', 'vf-pending', 'vf-invalid-validate'])
+  })
+
+  it('should show a field message if input is invalid', () => {
+    wrapper.setData({
+      state: {
+        $touched: true,
+        $invalid: true
+      }
+    })
+    expect(wrapper.contains('div.form-control-feedback')).to.equal(true)
+
+    const fieldMessageElement = wrapper.find('div.form-control-feedback')
+    expect(fieldMessageElement.text()).to.equal('This field is required')
+  })
+
+  describe('TypedFieldComponent with type number', () => {
+    const field = {
+      id: 'typed-field',
+      label: 'Typed Field',
+      description: 'This is a field that supports many types',
+      type: 'number',
+      visible: true,
+      required: true,
+      disabled: false,
+      validators: []
+    }
+
+    const mockParentFunction = () => {
```

Comment:
duplicate code , general setup can be moved to module head or some setup function, if the test needs to be updated/refactored duplicate code is a problem.